### ISSUE TITLE
Throw an Error object instead of a string

### ISSUE THREAD

**Issue by scottnonnenberg** [matankb/fetch-reject#2]
`throw response.statusText;` ([here](https://github.com/matankb/fetch-reject/blob/ccca47de7dfd57628e64f0b9ccbf3c5a972f9e4f/index.js#L13)) should be something like `throw new Error(response.statusText)`. 

You might also consider adding more information to the error like `http-errors` [does](https://github.com/jshttp/http-errors/blob/1a826d7ac31dde16931b9c566041697939ebd0e0/index.js#L81): a `status` key with the HTTP code. 

Background: http://www.devthought.com/2011/12/22/a-string-is-not-an-error/

**Comment by matankb** [comment]
Good point about throwing an `Error()` object. I definitely agree that should be done. 

As for adding more information - I think the best way to implement this would be to attach the status key to the error object:

``` javascript
let error = new Error(response.statusText);
error.status = response.status;
throw error;
```

That way, it'll be backward-compatible with existing code that expects the thrown `error` to be a string while allowing new code to take advantage of the status.

Thanks for bringing this up - I'll implement it as soon as I can.

**Comment by matankb** [comment]
Fixed in 4a7dc07
